src/main/java/io/aiven/kafka/connect/opensearch/spi/OpensearchClientConfigurator.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019 Aiven Oy
+ * Copyright 2016 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.opensearch.spi;
+
+import io.aiven.kafka.connect.opensearch.OpensearchSinkConnectorConfig;
+
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+
+/**
+ * The extension points to allow OpensearchClient customization based on the the provided configuration.
+ */
+public interface OpensearchClientConfigurator {
```

Comment:
Thoughts on adding a method to add configuration properties to the connector's `ConfigDef`, so that programmatic UIs and other tools that use the REST API to learn about a connector's configuration surface can be aware of its auth-related properties?

Thinking of something like this:
```java
    void addConfigs(ConfigDef configDef);
```

Reply:
That would be next step - extensible properties (so we could add cloud specific settings dynamically and apply the configurators at runtime).

Reply:
👍 Good call, can leave this as a next step.